src/ui/backend.rs

```diff
@@ -11,6 +11,65 @@ use crate::{
     validator::ErrorMessage,
 };
 
+macro_rules! __print_impl {
+    (
+        $self:ident,
+        $val_expr:expr,
+        $write_fn:ident,
+        $val_ident:ident => $val_content:expr,
+        $line_ident:ident => $converter:expr
+    ) => {{
+        let $val_ident = $val_expr;
+        let __content = $val_content;
+        let __converted = newline_converter::unix2dos(&__content);
+
+        for $line_ident in __converted.split_inclusive("\r\n") {
+            if $self.is_new_line {
+                $self
+                    .terminal
+                    .cursor_move_to_column($self.render_config.global_indentation)?;
```

Comment:
Found a tiny bug* where we would leave the cursor at the end of the prompt in column 0, not respecting the global_indentation.

\* It isn't really a bug, but a decision we'll have to make, take a look at the example modifications I made at this commit 43da960892af061394b7e5f3fcb37ff9b7aefade

Which one of the following behaviors do we want a "println!" after the prompt to have?

```
╔══════════════════════════════════════════════════════╗
║ > Description: mikael
║ Your transaction has been successfully recorded.
```

or

```
╔══════════════════════════════════════════════════════╗
║ > Description: mikael
Your transaction has been successfully recorded.
```

In other words, should we leave the cursor at column 0 or an indented value after the prompt exits?

Your PR is currently leaving it at position 0, I did a little refactoring on 43da960892af061394b7e5f3fcb37ff9b7aefade that left it at the indented column, what do you think?